libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp

```diff
@@ -480,15 +487,23 @@ void AP_RCProtocol_CRSF::process_link_stats_rx_frame(const void* data)
 {
     const LinkStatisticsRXFrame* link = (const LinkStatisticsRXFrame*)data;
 
-    _link_status.rssi = link->rssi_percent * 255.0f * 0.01f;
+    if (_use_lq_for_rssi) {
+        _link_status.rssi = derive_scaled_lq_value(link->link_quality);
+    } else {
+        _link_status.rssi = link->rssi_percent * 255.0f * 0.01f;
+    }
 }
 
 // process link statistics to get TX RSSI
 void AP_RCProtocol_CRSF::process_link_stats_tx_frame(const void* data)
 {
     const LinkStatisticsTXFrame* link = (const LinkStatisticsTXFrame*)data;
 
-    _link_status.rssi = link->rssi_percent * 255.0f * 0.01f;
+    if (_use_lq_for_rssi) {
+        _link_status.rssi = derive_scaled_lq_value(link->link_quality);
```

Comment:
As far as I can tell, the existing V3 status frame decode functions both update `_link_status.rssi` as there is only one RSSI and one Link Quality argument being passed by the class (see below). I have just kept with that behaviour (apart from it would be LQ not RSSI if this new option is set).

Ideally AP_RCProtocol_Backend::add_input() would allow for more data types (or some other mechanism for RC protocols with additional status fields).

- AP_RCProtocol_CRSF.h:
```
struct LinkStatus {
    int16_t rssi = -1;
    int16_t link_quality = -1;
    RFMode rf_mode;
};
...
volatile struct LinkStatus _link_status;
``` 
- AP_RCProtocol_CRSF.cpp:
```
if (decode_crsf_packet()) {
    add_input(MAX_CHANNELS, _channels, false, _link_status.rssi, _link_status.link_quality);
}
```